src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -5,72 +5,61 @@
 import { commonOptions, languageServerOptions } from '../shared/options';
 import { existsSync } from 'fs';
 import { CSharpExtensionId } from '../constants/csharpExtensionId';
-import { getDotnetInfo } from '../shared/utils/getDotnetInfo';
 import { readFile } from 'fs/promises';
-import { RuntimeInfo } from '../shared/utils/dotnetInfo';
+import { IDotnetFindPathContext } from './dotnetRuntimeExtensionApi';
 
 export const DotNetRuntimeVersion = '8.0.10';
 
-interface IDotnetAcquireResult {
-    dotnetPath: string;
-}
-
 /**
  * Resolves the dotnet runtime for a server executable from given options and the dotnet runtime VSCode extension.
  */
 export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
     constructor(
         private platformInfo: PlatformInformation,
-        /**
-         * This is a function instead of a string because the server path can change while the extension is active (when the option changes).
-         */
-        private getServerPath: (platform: PlatformInformation) => string,
         private channel: vscode.OutputChannel,
         private extensionPath: string
     ) {}
 
     private hostInfo: HostExecutableInformation | undefined;
 
     async getHostExecutableInfo(): Promise<HostExecutableInformation> {
-        let dotnetRuntimePath = commonOptions.dotnetPath;
-        const serverPath = this.getServerPath(this.platformInfo);
-
-        // Check if we can find a valid dotnet from dotnet --version on the PATH.
-        if (!dotnetRuntimePath) {
-            const dotnetPath = await this.findDotnetFromPath();
-            if (dotnetPath) {
-                return {
-                    version: '' /* We don't need to know the version - we've already verified its high enough */,
-                    path: dotnetPath,
-                    env: this.getEnvironmentVariables(dotnetPath),
-                };
+        let dotnetExecutablePath: string;
+        if (commonOptions.dotnetPath) {
+            const dotnetExecutableName = this.getDotnetExecutableName();
+            dotnetExecutablePath = path.join(commonOptions.dotnetPath, dotnetExecutableName);
+        } else {
+            if (this.hostInfo) {
+                return this.hostInfo;
             }
-        }
-
-        // We didn't find it on the path, see if we can install the correct runtime using the runtime extension.
-        if (!dotnetRuntimePath) {
-            const dotnetInfo = await this.acquireDotNetProcessDependencies(serverPath);
-            dotnetRuntimePath = path.dirname(dotnetInfo.path);
-        }
 
-        const dotnetExecutableName = this.getDotnetExecutableName();
-        const dotnetExecutablePath = path.join(dotnetRuntimePath, dotnetExecutableName);
-        if (!existsSync(dotnetExecutablePath)) {
-            throw new Error(`Cannot find dotnet path '${dotnetExecutablePath}'`);
+            this.channel.appendLine(`Acquiring .NET runtime version ${DotNetRuntimeVersion}`);
```

Comment:
This might give the impression we are always pulling down a runtime. Perhaps
```suggestion
            this.channel.appendLine(`Locating .NET runtime version ${DotNetRuntimeVersion}`);
```